tst/style/cpplint.py

```diff
@@ -49,7 +49,6 @@
 import math  # for log
 import os
 import re
-import sre_compile
```

Comment:
Got deprecation warnings from `cpplint.py` so I removed this.

Reply:
Is it worth bumping the version of `cpplint`? This one was addressed [here](https://github.com/cpplint/cpplint/pull/214)

Reply:
:+1: Done. Updated `cpplint` to newest version. Note:
- I had to disable the check for using the directory for include files. Parthenon simply does not follow this convention.
- I had to disable the check for indent in a namespace, as the newest version cpplint erroneously finds multiline function prototypes as indents. It would be impossible to square with `clang-format` formatting.

Reply:
For whatever reason the new version of cpplint is much more aggressive about missed headers. A fresh sweep adding include statements was required.